src/pools/thread/dynamic.ts

```diff
@@ -31,6 +36,11 @@ export class DynamicThreadPool<
     super(min, filename, opts)
   }
 
+  /**
+   * Choose a thread in a round robin fashion.
+   *
+   * If all active threads are currently busy, try creating a new thread and using it.
```

Comment:
```suggestion
   * Choose a thread for the next task.
   *
   * It will first check for and return an idle thread.
   * If all threads are busy, then it will try to create a new one up to the `max` thread count.
   * If the max thread count is reached, the emitter will emit a `FullPool` event and it will fall back to using a round robin algorithm to distribute the load.
```